src/lib/input/input-container.scss

```diff
@@ -3,70 +3,124 @@
+$mat-input-prefix-suffix-icon-font-scale: 1.5 !default;
+
+// The line-height for all text in the md-input-container
+$mat-input-line-height: 1 + $mat-input-line-spacing;
+// The padding on the infix. Mocks show half of the text size, but seem to measure from the edge of
+// the text itself, not the edge of the line; therefore we subtract off 1/2 of the line spacing.
+$mat-input-infix-padding: 0.5em - $mat-input-line-spacing / 2;
+// The margin applied to the input-infix to reserve space for the floating label.
+$mat-input-infix-margin-top: 1em * $mat-input-line-height * $mat-input-helper-font-scale;
+// Font size to use for the label and subscript text.
+$mat-input-helper-font-size: $mat-input-helper-font-scale * 100%;
+// Font size to use for the for the prefix and suffix icons.
+$mat-input-prefix-suffix-icon-font-size: $mat-input-prefix-suffix-icon-font-scale * 100%;
+// The space between the bottom of the input table and the subscript container. Mocks show half of
+// the text size, but this margin is applied to an element with the helper text font size, so we
+// need to divide by the scale factor to make it half of the original text size. We again need to
+// subtract off 1/2 of the line spacing since the mocks measure to the edge of the text, not the
+// edge of the line.
+$mat-input-subscript-margin-top: 0.5em / $mat-input-helper-font-scale - $mat-input-line-spacing / 2;
+// The padding applied to the input-wrapper to reserve space for the subscript, since it's
+// absolutely positioned. This is a combination of the subscript's margin and line-height, but we
+// need to multiply by the helper font scale factor since the wrapper has a larger font size.
+$mat-input-wrapper-padding-bottom:
+    ($mat-input-subscript-margin-top + $mat-input-line-height) * $mat-input-helper-font-scale;
+// The amount of y-offset to apply to the placeholder when it is floating.
+$mat-input-floating-placeholder-offset: -$mat-input-infix-padding - 1em;
 
-// Gradient for showing the dashed line when the input is disabled.
-$mat-input-underline-disabled-background-image:
-    linear-gradient(to right, rgba(0, 0, 0, 0.26) 0%, rgba(0, 0, 0, 0.26) 33%, transparent 0%);
 
 // Applies a floating placeholder above the input itself.
 @mixin mat-input-placeholder-floating {
   display: block;
-  transform: translate3d(0, -1.35em, 0) scale($mat-input-floating-placeholder-scale-factor);
-  width: 100% / $mat-input-floating-placeholder-scale-factor;
+  transform: translateY(-$mat-input-infix-margin-top - $mat-input-infix-padding)
+             perspective(1px) // fixes text blurriness.
+             scale($mat-input-helper-font-scale);
+  width: 100% / $mat-input-helper-font-scale;
 }
 
+
 .mat-input-container {
   display: inline-block;
   position: relative;
   font-family: $mat-font-family;
-  line-height: normal;
+
+  // TODO(mmalerba): should this come from typography settings? If so a lot of this CSS needs to go
```

Comment:
I'm not sure whether we'd be able to exactly match the spec if we went unitless. That being said, not all values have to go through the typogaphy API necessarily. You can take the font size and have a custom line height.